src/ParameterParser.h

```diff
@@ -396,72 +417,110 @@ namespace snowcrash {
             }
 
-            // Remove param name
-            innerSignature = innerSignature.substr(firstSpace + 1);
-            size_t descriptionPos = innerSignature.find(snowcrash::DescriptionIdentifier);
+            std::string firstChar = innerSignature.substr(0, 1);
 
-            // Remove description
-            if (descriptionPos != std::string::npos) {
-                innerSignature = innerSignature.substr(0, descriptionPos);
-                innerSignature = TrimString(innerSignature);
+            // If first char is ':' or `=`
+            if (firstChar == ":" || firstChar == "=") {
+
+                innerSignature = innerSignature.substr(1);
+                TrimString(innerSignature);
+
+                if (innerSignature.empty()) {
+                    return NotParameterType; // No sample or default value
+                }
+
+                // Traverse over the value
+                while (!innerSignature.empty()) {
+
+                    std::string first = innerSignature.substr(0, 1);
+
+                    if (first == "`") {
+                        RetrieveEscaped(innerSignature);
+                        TrimString(innerSignature);
+                    }
+                    else if (first == "(") {
+                        break;
+                    }
+                    else {
+                        innerSignature = innerSignature.substr(1);
+                    }
+                }
+
+                return firstChar == ":" ? NewParameterType : OldParameterType;
             }
 
-            size_t attributesPos = innerSignature.find("(");
+            if (innerSignature.substr(0, 1) == "(") {
 
-            if (attributesPos != std::string::npos) {
+                // We should use `matchBrackets` if the parameters are supported to be more complex
                 size_t endOfAttributesPos = innerSignature.find_last_of(")");
 
                 if (endOfAttributesPos == std::string::npos) {
-                    return false; // Expecting close of attributes
+                    return NotParameterType; // Expecting close of attributes
                 }
 
-                // Remove attributes
-                innerSignature = innerSignature.substr(0, attributesPos);
-                innerSignature = TrimString(innerSignature);
-            }
+                std::string attributes = innerSignature.substr(1, endOfAttributesPos);
 
-            if (innerSignature.length() == 0) {
-                return true;
-            }
-
-            if (innerSignature.substr(0,1) == "=") {
-                innerSignature = innerSignature.substr(1);
-                innerSignature = TrimString(innerSignature);
+                if (RegexMatch(attributes, "enum\\[[^][]+]")) {
+                    return NewParameterType;
+                }
 
-                if (innerSignature.length() == 0) {
-                    return false; // No default value
+                if (RegexMatch(attributes, "`")) {
```

Comment:
Yes. Again went with regex because this is similar to the regex searching for `enum` above.